### ISSUE TITLE
v0.84.x:  Panic: interface conversion: interface {} is map[string]string, not maps.Params

### ISSUE THREAD

**Issue by SirUli** [gohugoio/hugo#8709]
### What version of Hugo are you using (`hugo version`)?

<pre>
0.84.2 Extended

</pre>

### Does this issue reproduce with the latest release?
Yes

### Issue description:

As in the closed issue #8679 i get also in v0.84.2 the following message when running ``hugo mod get -u``:

```
panic: interface conversion: interface {} is string, not maps.Params
goroutine 1 [running]:
github.com/gohugoio/hugo/config.(*defaultConfigProvider).getNestedKeyAndMap(0xc00045eff0, 0xc0004f5b50, 0xc, 0x1, 0xc, 0x2f7cca0, 0x0)
	/root/project/hugo/config/defaultConfigProvider.go:425 +0x38a
github.com/gohugoio/hugo/config.(*defaultConfigProvider).Set(0xc00045eff0, 0xc0004f5b50, 0xc, 0x1c36540, 0xc0002bc880)
	/root/project/hugo/config/defaultConfigProvider.go:171 +0x139
github.com/gohugoio/hugo/hugolib.configLoader.applyOsEnvOverrides(0x2366170, 0xc00045eff0, 0x2365680, 0x2fc0d60, 0x2368798, 0xc0004f1c20, 0x0, 0x0, 0xc000050094, 0x5, ...)
	/root/project/hugo/hugolib/config.go:346 +0x494
github.com/gohugoio/hugo/hugolib.LoadConfig(0x2365680, 0x2fc0d60, 0x2368798, 0xc0004f1c20, 0x0, 0x0, 0xc000050094, 0x5, 0xc000050094, 0x5, ...)
	/root/project/hugo/hugolib/config.go:118 +0x790
github.com/gohugoio/hugo/commands.(*commandeer).loadConfig(0xc0004ef180, 0xc0004e0000, 0x2333000, 0xc000010018)
	/root/project/hugo/commands/commandeer.go:298 +0x385
github.com/gohugoio/hugo/commands.newCommandeer(0x7f4fdf1e0000, 0xc000420a90, 0x23303a0, 0xc0005e84a8, 0x0, 0x0, 0x0, 0x0, 0x98, 0xed00000001e9fa20, ...)
	/root/project/hugo/commands/commandeer.go:182 +0x257
github.com/gohugoio/hugo/commands.initializeConfig(0x1f20000, 0xc000420a90, 0x23303a0, 0xc0005e84a8, 0x0, 0x6441b6, 0x1f2e780, 0x5)
	/root/project/hugo/commands/hugo.go:121 +0x76
github.com/gohugoio/hugo/commands.(*modCmd).initConfig(0xc0005e84a8, 0x0, 0x1f2a512, 0x4, 0xc0000ba320)
	/root/project/hugo/commands/mod.go:287 +0x65
github.com/gohugoio/hugo/commands.(*modCmd).withModsClient(0xc0005e84a8, 0xc00073fc00, 0xc00073fc98, 0xc0000bc900, 0xc00073fce8)
	/root/project/hugo/commands/mod.go:269 +0x38
github.com/gohugoio/hugo/commands.(*commandsBuilder).newModCmd.func1(0xc000213400, 0xc000381e80, 0x1, 0x1, 0x0, 0x0)
	/root/project/hugo/commands/mod.go:199 +0x245
github.com/spf13/cobra.(*Command).execute(0xc000213400, 0xc000381e80, 0x1, 0x1, 0xc000213400, 0xc000381e80)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003b9180, 0xc000381cc0, 0x5, 0x6)
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/gohugoio/hugo/commands.Execute(0xc00003c090, 0x3, 0x3, 0x4ce1c5, 0xc0000481d8, 0x1997000, 0x0)
	/root/project/hugo/commands/hugo.go:88 +0xb9
main.main()
	/root/project/hugo/main.go:23 +0x76
script returned exit code 2
```

Anything i can provide to get this resolved? Unfortunately the issue came up in v0.84.0 and persisted throughout v0.84.1 and v0.84.2. It works with the previous v0.83.1.

Best Regards, Uli

**Comment by bep** [comment]
>Anything i can provide to get this resolved? 

A project demonstrating the problem.

**Comment by SirUli** [comment]
@bep sorry wasn't fast enough. You rock! That fixed it - i can confirm that 0.84.3 works.
